fix(add-app-overlay): Change design of create an application modal

vikram-raj opened the pull request:
Story: https://openshift.io/openshiftio/Openshift_io/plan/detail/978

christianvogt commented on src/app/space/add-app-overlay/add-app-overlay.component.less:
```diff
@@ -1,165 +1,46 @@
+  .required {
+    color: @color-pf-red;
+  }
+  &--container {
```
Elements should be named with double underscore instead of double dash.

christianvogt commented on src/app/app.component.html:
```diff
@@ -6,7 +6,13 @@
 </pfng-toast-notification-list>
 <f8-add-space-overlay class="launcher fade in" *ngIf="showAddSpaceOverlay"></f8-add-space-overlay>
-<f8-add-app-overlay [preselectedFlow]="addAppFlow" class="code-imports fade in" *ngIf="showAddAppOverlay"></f8-add-app-overlay>
+<div class="modal fade" bsModal #modalAddAppOverlay="bs-modal" tabindex="-1" role="dialog" aria-hidden="true">
```
This functionality should be placed in a separate component. The app component is being overloaded with extra concerns that it shouldn't need to worry about.
Same could be said about the `f8-add-space-overlay`. The controlling of opening the overlay shouldn't be the concern of the app but rather the overlay component, or a wrapper thereof.

The modal markup should be placed in a separate component such as `add-app-overlay-modal`.

That being said, since the existing functionality was in the app. I'm ok for now to accept this and track refactoring both in another issue.

joshuawilson replied:
I completely agree. The app component is the entry point and should just be calling the highest level component or components.

You have added a `<div>` with a modal to the highest parent in the UI. This should be fixed before it is merged. This could introduce unforeseen problems.

vikram-raj replied:
Done in https://github.com/fabric8-ui/fabric8-ui/pull/3427/commits/bb3764fe8083dcddf92b231d1c71f8cac451e981

christianvogt commented on src/app/app.component.less:
```diff
@@ -1,3 +1,7 @@
 }
+
+.f8-modal-lg {
+  width: 70%;
```
Please change this to:
```
width: 1500px;
max-width: calc(100% - 20px);
```
This way the size stays responsive.
The `-20px` accounts for the modal margins.
`1500px` is somewhat arbitrary as it looks good for large desktop screens and scales accordingly with the above css change.

christianvogt commented on src/app/space/add-app-overlay/add-app-overlay.component.html:
```diff
@@ -1,150 +1,120 @@
+            Learn more <i class="fa fa-external-link"></i>
+          </a>
+        </p>
+        <h5>What is a codebase?</h5>
```
Add a `<br/>` before this `<h5>`

christianvogt commented on src/app/space/add-app-overlay/add-app-overlay.component.html:
```diff
@@ -1,150 +1,120 @@
+            Learn more <i class="fa fa-external-link"></i>
+          </a>
+        </p>
+        <h5>When I import a codebase?</h5>
```
Add a `<br/>` before this `<h5>`

christianvogt commented:
Please remove all the `font-size-small`. We should be using the standard font-size.
Except for the error message under the name input field.

christianvogt commented:
When using bootstrap `col-` classes, columns should always be contained within a parent with a `row`. This will remove the extra padding on the right and left of the row as well as separate vertically the rows. If you make this change, it could affect the padding styles you've already made.

vikram-raj commented:
@christianvogt I addressed all the comments in https://github.com/fabric8-ui/fabric8-ui/pull/3427/commits/4e40de46dcdba5f6391a086b6eb971acd5b1bbb6. Please review it again.

christianvogt commented:
\@ppitonak @vikram-raj this change seems to break the e2e. Looks like a valid failure. We will need to update the e2e tests.

vikram-raj commented:
@christianvogt e2e test has been fixed.

joshuawilson commented on src/app/app.component.spec.ts:
```diff
@@ -20,6 +20,7 @@ import {
   Logger
 } from 'ngx-base';
 import { BsModalService } from 'ngx-bootstrap/modal';
+import { ModalModule } from 'ngx-bootstrap/modal';
```
Why is this on a new line? It should be added to the import above.

vikram-raj replied:
Because of auto import. I will fix this.

joshuawilson commented on src/app/app.component.ts:
```diff
@@ -3,6 +3,7 @@ import { Title } from '@angular/platform-browser';
 import { ActivatedRoute, Event, NavigationEnd, NavigationError, Router } from '@angular/router';
 import { Broadcaster, Logger } from 'ngx-base';
 import { BsModalService } from 'ngx-bootstrap/modal';
+import { ModalDirective } from 'ngx-bootstrap/modal';
```
Why is this on a new line? It should be added to the import above.

vikram-raj replied:
Because of auto import. I will fix this.

joshuawilson commented on src/app/app.component.ts:
```diff
@@ -41,6 +42,7 @@ export class AppComponent {
   protected subscriptions: Subscription[] = [];
 
   @ViewChild('connectToGithubModal') connectToGithubModal: TemplateRef<any>;
+  @ViewChild('modalAddAppOverlay') modalAddAppOverlay: ModalDirective;
```
Along with the changes in the html, this should NOT be here either. It belongs in a child.

joshuawilson commented on src/app/space/add-app-overlay/add-app-overlay.component.html:
```diff
@@ -1,150 +1,120 @@
+      <p>
+        A codebase is a generic representation of an application's code.
+        Openshift.io also stores application metadata for version control.
+        Since Openshift.io provides a hosted instance of Eclipse Che, you can
```
Please change `Openshift.io` to `CodeReady ToolChain`

joshuawilson commented on src/app/space/add-app-overlay/add-app-overlay.component.html:
```diff
@@ -1,150 +1,120 @@
+        Openshift.io also stores application metadata for version control.
+        Since Openshift.io provides a hosted instance of Eclipse Che, you can
+        create a workspace for any application and use the browser-based IDE
+        and containerized development/testing enviroment to make updates to your codebase.
```
`enviroment` is misspelled

joshuawilson commented on src/app/space/add-app-overlay/add-app-overlay.component.html:
```diff
@@ -1,150 +1,120 @@
+      <p>
+        When you create an application in Openshift.io, a new codebase
+        is created and stored in your git repository. Openshift.io also
+        generates a new pipeline and pipeline enviroments for the application
```
`enviroments` is misspelled

joshuawilson commented on src/app/space/add-app-overlay/add-app-overlay.component.html:
```diff
@@ -1,150 +1,120 @@
+      <h5>What happens when I create an application?</h5>
+      <p>
+        When you create an application in Openshift.io, a new codebase
+        is created and stored in your git repository. Openshift.io also
```
Please change `Openshift.io` to `CodeReady ToolChain`

joshuawilson commented:
@fabric8-ui/docs Please review. This may impact the documentation.

Preeticp commented:
@joshuawilson @christianvogt @nimishamukherjee @serenamarie125, currently we talk about what it means to create an app, then tell the user what is a codebase. We should have the codebases section on top, then the user can make a choice if she wants to create an app using available codebases or import an exisitng one. 
It does not make sense to tell the user that when you create an app a new codebase is added. We need to explain to the user what a codebase is first. 
I have raised this point in the invision too: https://redhat.invisionapp.com/share/E7O609F8T9W#/screens/332319128/comments/107649910

joshuawilson commented:
good question

fabric8cd commented:
:tada: This PR is included in version 3.0.0 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/fabric8-ui)
- [GitHub release](https://github.com/fabric8-ui/fabric8-ui/releases/tag/v3.0.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
